doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -91,47 +255,72 @@ In this case, no spaces are allowed in the symbol.  Also, if the
 symbol starts with a double quote, it must be surrounded with single
 quotes to prevent confusion with the beginning of a string symbol.
 
-A notation binds a syntactic expression to a term, called its :gdef:`interpretation`. Unless the parser
-and pretty-printer of Coq already know how to deal with the syntactic
-expression (such as through :cmd:`Reserved Notation` or for notations
-that contain only literals), explicit precedences and
-associativity rules have to be given.
+Here are examples from the initial state of Coq.
```

Comment:
(I can't comment parts that did not change but this seems important).
The example above does not work for me (maybe I misunderstood):
```Coq
Reserved Notation "A ""I'm an """"infix"""" symbol"" B" (at level 199, right associativity).
```
gives me:
```
String '"I'm an ""infix"" symbol"' now a keyword

Syntax Error: Lexer: Bad token "\"I'm an \"\"infix\"\" symbol\""
```

Reply:
I don't know, tried a few things but didn't manage to make it work. Maybe @herbelin would have an idea?
In any case, this should be transformed to an actual example with the coqtop syntax to avoid breaking it by accident.

Reply:
Looks like strings with spaces are not allowed as keywords, so they only work when used in a position which doesn't force a keyword.
For instance https://github.com/coq/coq/blob/a8ef679aedb5860a9817e0b8172d9f9cb69d0fda/test-suite/output/NotationSyntax.v#L29-L30 (keyword string with no space) works, but breaks if we add a space,
and https://github.com/coq/coq/blob/a8ef679aedb5860a9817e0b8172d9f9cb69d0fda/test-suite/output/NotationSyntax.v#L24-L25 (non keyword) works even if we add a space

This is probably a bug.